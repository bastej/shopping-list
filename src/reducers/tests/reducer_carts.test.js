import cartReducer from "../reducer_carts";

import {
  CREATE_NEW_CART,
  UPDATE_CART_PRODUCT_COUNT,
  DELETE_CART_PRODUCT
} from "../../actions/types";

describe("CardReducer", () => {
  const initialState = {
    0: {
      id: 0,
      productsList: {
        0: {
          id: 0,
          count: 10
        },
        1: {
          id: 1,
          count: 3
        }
      }
    }
  };

  it("returns the initial state", () => {
    expect(cartReducer(undefined, {})).toMatchSnapshot();
  });

  it("handles the createNewCart action", () => {
    expect(
      cartReducer({}, { type: CREATE_NEW_CART, payload: {} })
    ).toMatchSnapshot();
  });

  describe("updateProductCount", () => {
    it("should increment product count + 1", () => {
      expect(
        cartReducer(initialState, {
          type: UPDATE_CART_PRODUCT_COUNT,
          payload: {
            actionType: "increment",
            listID: 0,
            productID: 0
          }
        })
      ).toEqual({
        0: {
          id: 0,
          productsList: {
            0: {
              id: 0,
              count: 11
            },
            1: {
              id: 1,
              count: 3
            }
          }
        }
      });
    });

    it("should decrement product count - 1", () => {
      expect(
        cartReducer(initialState, {
          type: UPDATE_CART_PRODUCT_COUNT,
          payload: {
            actionType: "",
            listID: 0,
            productID: 0
          }
        })
      ).toEqual({
        0: {
          id: 0,
          productsList: {
            0: {
              id: 0,
              count: 9
            },
            1: {
              id: 1,
              count: 3
            }
          }
        }
      });
    });

    it("should delete product", () => {
      expect(
        cartReducer(initialState, {
          type: DELETE_CART_PRODUCT,
          payload: {
            listID: 0,
            productID: 0
          }
        })
      ).toEqual({
        0: {
          id: 0,
          productsList: {
            1: {
              id: 1,
              count: 3
            }
          }
        }
      });
    });
  });
});
