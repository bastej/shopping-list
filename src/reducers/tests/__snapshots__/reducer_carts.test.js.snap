// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardReducer handles the createNewCart action 1`] = `
Object {
  "0": Object {
    "id": 0,
  },
}
`;

exports[`CardReducer returns the initial state 1`] = `
Object {
  "0": Object {
    "createDate": "04.09.2018",
    "id": 0,
    "productsList": Object {
      "0": Object {
        "calories": 161,
        "carbohydrates": 36.6,
        "category": "vegetables",
        "count": 10,
        "fats": 0.2,
        "id": 0,
        "name": "potato",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/752_thumb.jpg",
        "proteins": 4.3,
        "serving_weight_grams": 173,
        "tags": Array [
          "Cholesterol Free",
          "Fat Free",
          "Good Source of Fiber",
          "Saturated Fat Free",
          "Very Low Sodium",
        ],
      },
      "1": Object {
        "calories": 468,
        "carbohydrates": 0,
        "category": "meat",
        "count": 1,
        "fats": 28,
        "id": 1,
        "name": "salmon",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/375_thumb.jpg",
        "proteins": 50.2,
        "serving_weight_grams": 227,
        "tags": Array [
          "Atkins 20",
          "Atkins 40",
          "Atkins 100",
          "Atkins Friendly",
          "Low Carb",
          "High Protein",
          "Very Low Sodium",
          "Sugar Free",
        ],
      },
      "2": Object {
        "calories": 452,
        "carbohydrates": 3.6,
        "category": "dairy",
        "count": 3,
        "fats": 37.2,
        "id": 2,
        "name": "cheese",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/1034_thumb.jpg",
        "proteins": 25.6,
        "serving_weight_grams": 112,
        "tags": Array [
          "Atkins 20",
          "Atkins 40",
          "Atkins 100",
          "Atkins Friendly",
          "Low Carb",
          "Ketogenic Diet",
          "Good Source of Protein",
          "Sugar Free",
        ],
      },
      "3": Object {
        "calories": 41,
        "carbohydrates": 9.5,
        "category": "vegetables",
        "count": 5,
        "fats": 0.2,
        "id": 3,
        "name": "onion",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/717_thumb.jpg",
        "proteins": 1.3,
        "serving_weight_grams": 94,
        "tags": Array [
          "Low Calorie",
          "Cholesterol Free",
          "Fat Free",
          "Saturated Fat Free",
          "Sodium Free",
        ],
      },
    },
    "title": "Christmas shopping",
    "type": "cart",
  },
  "1": Object {
    "createDate": "21.11.2018",
    "id": 1,
    "productsList": Object {
      "0": Object {
        "calories": 16,
        "carbohydrates": 4.2,
        "category": "sweets",
        "count": 4,
        "fats": 0,
        "id": 0,
        "name": "sugar",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/834_thumb.jpg",
        "proteins": 0,
        "serving_weight_grams": 4.2,
        "tags": Array [
          "Cholesterol Free",
          "Fat Free",
          "Low Saturated Fat",
          "Sodium Free",
        ],
      },
      "1": Object {
        "calories": 124,
        "carbohydrates": 0,
        "category": "other",
        "count": 4,
        "fats": 14,
        "id": 1,
        "name": "oil",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/1031_thumb.jpg",
        "proteins": 0,
        "serving_weight_grams": 14,
        "tags": Array [
          "Atkins 20",
          "Atkins 40",
          "Atkins 100",
          "Atkins Friendly",
          "Low Carb",
          "Cholesterol Free",
          "Ketogenic Diet",
          "Low Saturated Fat",
          "Sodium Free",
          "Sugar Free",
        ],
      },
      "2": Object {
        "calories": 455,
        "carbohydrates": 95.4,
        "category": "other",
        "count": 14,
        "fats": 1.2,
        "id": 2,
        "name": "flour",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/206_thumb.jpg",
        "proteins": 12.9,
        "serving_weight_grams": 125,
        "tags": Array [
          "Cholesterol Free",
          "Low Carb",
          "Good Source of Fiber",
          "High Protein",
          "Low Saturated Fat",
          "Sodium Free",
          "Sugar Free",
        ],
      },
    },
    "title": "For kitchen",
    "type": "cart",
  },
  "2": Object {
    "createDate": "1.11.2008",
    "id": 2,
    "productsList": Object {
      "0": Object {
        "calories": 105,
        "carbohydrates": 27,
        "category": "fruits",
        "count": 2,
        "fats": 0.4,
        "id": 0,
        "name": "banana",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/399_thumb.jpg",
        "proteins": 1.3,
        "serving_weight_grams": 118,
        "tags": Array [
          "Cholesterol Free",
          "Fat Free",
          "Good Source of Fiber",
          "Saturated Fat Free",
          "Sodium Free",
        ],
      },
      "1": Object {
        "calories": 122,
        "carbohydrates": 11.7,
        "category": "dairy",
        "count": 1,
        "fats": 4.8,
        "id": 1,
        "name": "2% milk",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/9337_thumb.jpg",
        "proteins": 8.1,
        "serving_weight_grams": 244,
        "tags": Array [
          "Low Carb",
          "Low Cholesterol",
          "Low Fat",
          "Good Source of Protein",
          "Very Low Sodium",
        ],
      },
      "2": Object {
        "calories": 153,
        "carbohydrates": 27.4,
        "category": "other",
        "count": 1,
        "fats": 2.6,
        "id": 2,
        "name": "oat",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/712_thumb.jpg",
        "proteins": 5.3,
        "serving_weight_grams": 40.5,
        "tags": Array [
          "Cholesterol Free",
          "Good Source of Fiber",
          "Good Source of Protein",
          "Low Saturated Fat",
          "Sodium Free",
          "Sugar Free",
        ],
      },
      "3": Object {
        "calories": 72,
        "carbohydrates": 0.4,
        "category": "dairy",
        "count": 2,
        "fats": 4.8,
        "id": 3,
        "name": "eggs",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/775_thumb.jpg",
        "proteins": 6.3,
        "serving_weight_grams": 50,
        "tags": Array [
          "Atkins 20",
          "Atkins 40",
          "Atkins 100",
          "Atkins Friendly",
          "Low Carb",
          "Good Source of Protein",
          "Low Sodium",
          "Sugar Free",
        ],
      },
      "4": Object {
        "calories": 107,
        "carbohydrates": 3.7,
        "category": "dairy",
        "count": 2,
        "fats": 4.7,
        "id": 4,
        "name": "cottage cheese",
        "photo": "https://d2xdmhkmkbyw75.cloudfront.net/510_thumb.jpg",
        "proteins": 12.1,
        "serving_weight_grams": 108.75,
        "tags": Array [
          "Atkins 100",
          "Atkins Friendly",
          "Low Carb",
          "Low Cholesterol",
          "Low Fat",
          "High Protein",
        ],
      },
    },
    "title": "Protein shake",
    "type": "cart",
  },
}
`;
